integration-tests/velocity/src/test/java/org/apache/camel/quarkus/component/velocity/it/VelocityTest.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.quarkus.component.velocity.it;
+
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Map;
+
+import io.quarkus.test.junit.QuarkusTest;
+import io.restassured.RestAssured;
+import io.restassured.http.ContentType;
+import org.junit.jupiter.api.Test;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+@QuarkusTest
+class VelocityTest {
+
+    public static final String OLD_BODY = "old_body";
+    public static final String BODY = "bar";
+    public static final String NEW_BODY = "new_body";
+    public static final String MSG = "Dear Sheldon\n" +
+            "\n" +
+            "Thanks for the order of Camel in Action.\n" +
+            "\n" +
+            "Regards Apache Camel Riders Bookstore\n" +
+            "PS: Next beer is on me";
+
+    @Test
+    public void testTemplateViaHeader() throws Exception {
```

Comment:
Unless I missed something, I don't see any testing for what is likely to be the most common use case - reading a template from the classpath. 

Also, for the other templating components, we have some extra documentation which we should replicate for this extension.

https://camel.apache.org/camel-quarkus/latest/reference/extensions/mustache.html#_additional_camel_quarkus_configuration

Reply:
Actually I thought that use case with template via header, could contain more problems. 
But you are right, I'll add also basic use case with template from classpath.